src/mount/fuse/sfs_meta_fuse.cc

```diff
@@ -315,201 +324,200 @@ static uint32_t dir_metaentries_size(inode_t ino) {
+
+		if (dataBuffer + nameLength + kDirEntryMetaSize <= eptr) {
+			dataBuffer++;
+
+			if (nameLength > NAME_MAX - kDirEntryHexPreffixSize) {
+				inodeLength = NAME_MAX;
 			} else {
-				inoleng = nleng+9;
+				inodeLength = nameLength + kDirEntryHexPreffixSize;
 			}
-			put8bit(&buff,inoleng);
-			name = (const char*)dbuff;
-			dbuff+=nleng;
-			getINode(&dbuff, inode);
-			sprintf((char*)buff,"%08" PRIXiNode "|",inode);
-			if (nleng>255-9) {
-				memcpy(buff+9,name,255-9);
-				buff+=255;
+
+			put8bit(&buff, inodeLength);
+			name = dataBuffer;
+			dataBuffer += nameLength;
+			getINode(&dataBuffer, inode);
+			sprintf((char *)buff, "%0*" PRIXiNode "|", 2 * (int)sizeof(inode_t), inode);
+
+			if (nameLength > NAME_MAX - kDirEntryHexPreffixSize) {
+				memcpy(buff + kDirEntryHexPreffixSize, name, NAME_MAX - kDirEntryHexPreffixSize);
+				buff += NAME_MAX;
 			} else {
-				memcpy(buff+9,name,nleng);
-				buff+=9+nleng;
+				memcpy(buff + kDirEntryHexPreffixSize, name, nameLength);
+				buff += kDirEntryHexPreffixSize + nameLength;
 			}
-			putINode(&buff,inode);
-			put8bit(&buff,TYPE_FILE);
+
+			putINode(&buff, inode);
+			put8bit(&buff, TYPE_FILE);
 		} else {
-			safs_pretty_syslog(LOG_WARNING,"dir data malformed (trash)");
-			dbuff=eptr;
+			safs::log_warn("dir data malformed (trash)");
+			dataBuffer = eptr;
 		}
 	}
 }
 
-
-static void dirbuf_meta_fill(dirbuf *b, inode_t ino) {
+static void fillDirectoryBufferMeta(DirectoryBuffer *dirbuf, inode_t ino) {
 	int status;
-	uint32_t msize, dsize = 0, dcsize;
-	const uint8_t *dbuff = nullptr;
+	uint32_t metaEntriesSize, entriesSize = 0, dataEntriesSize;
+	const uint8_t *bufData = nullptr;
 
-	b->p = NULL;
-	b->size = 0;
-	msize = dir_metaentries_size(ino);
+	dirbuf->buffer.clear();
+	metaEntriesSize = getDirMetaEntriesSize(ino);
 
 	if (ino == SPECIAL_INODE_META_TRASH) {
-		status = fs_gettrash(&dbuff, &dsize);
-		if (status != SAUNAFS_STATUS_OK)
-			return;
-		dcsize = dir_dataentries_size(dbuff,dsize);
+		status = fs_gettrash(&bufData, &entriesSize);
+		if (status != SAUNAFS_STATUS_OK) { return; }
+		dataEntriesSize = getDirDataEntriesSize(bufData, entriesSize);
 	} else if (ino == SPECIAL_INODE_META_RESERVED) {
-		status = fs_getreserved(&dbuff, &dsize);
-		if (status != SAUNAFS_STATUS_OK)
-			return;
-		dcsize = dir_dataentries_size(dbuff, dsize);
+		status = fs_getreserved(&bufData, &entriesSize);
+		if (status != SAUNAFS_STATUS_OK) { return; }
+		dataEntriesSize = getDirDataEntriesSize(bufData, entriesSize);
 	} else {
-		dcsize = 0;
+		dataEntriesSize = 0;
 	}
 
-	if (msize + dcsize == 0)
-		return;
+	if (metaEntriesSize + dataEntriesSize == 0) { return; }
 
-	if (!(b->p = (uint8_t*)malloc(msize + dcsize))) {
-		safs_pretty_syslog(LOG_EMERG, "out of memory");
+	try {
+		dirbuf->buffer.resize(metaEntriesSize + dataEntriesSize);
+	} catch (const std::bad_alloc &e) {
+		safs::log_critical("fillDirectoryBufferMeta: Memory allocation failed: {}", e.what());
```

Comment:
In the future, consider using macro `__func__` if you want to log the name of the function.
Like this, the log line is shorter and if in the future, the function is renamed, you don't
have to worry about updating the log :)
```suggestion
		safs::log_critical("{}: Memory allocation failed: {}", __func__, e.what());
```

Reply:
Considered.